aexpect/session_ops.py

```diff
@@ -0,0 +1,598 @@
+    :type session: ShellSession
+    :param str expr: search expression
+    :param str path: file to search
+    :returns: whether there is a match or not
+    :rtype: bool
+    """
+    return grep_q(session, expr, path, flags=["r"])
+
+
+def grep_qvr(session, expr, path):
+    """
+    Invoke ``grep(1)`` on guest in silent mode searching for the complement of
+    *expr* recursively in *path*.
+
+    :param session: session to run the command on
+    :type session: ShellSession
+    :param str expr: search expression
+    :param str path: file to search
+    :returns: whether there is no match
+    :rtype: bool
+    """
+    return grep_q(session, expr, path, flags=["v", "r"])
+
+
+###############################################################################
+# stat(2)
+###############################################################################
+
+def stat(session, path, fmt=None):
+    """
+    Wrapper for ``stat``.
+
+    :param session: session to run the command on
+    :type session: ShellSession
+    :param str path: path to the filesystem entry to stat
+    :param str fmt: optional format passed to ``-d``
+    :returns: standard output of stat(1) on ``path``
+    :rtype: str
+    """
+    if fmt is None:
+        return session.cmd_output(r"stat '%s'" % path)
+
+    if not isinstance(fmt, str) or fmt[0] != "%":
+        raise RuntimeError("%s is not a valid format string for stat(1)" % fmt)
+
+    return session.cmd_output(r"stat -c %s '%s'" % (fmt, path))
+
+
+def get_atime(session, path):
+    """
+    Query the access time.
+
+    :param session: session to run the command on
+    :type session: ShellSession
+    :param str path: path to the filesystem entry to stat
+    :returns: standard output of stat(1) querying the access time of ``path``
+    :rtype: str
+    """
+    return stat(session, path, fmt="%X")
+
+
+def get_mtime(session, path):
+    """
+    Query the modification time.
+
+    :param session: session to run the command on
+    :type session: ShellSession
+    :param str path: path to the filesystem entry to stat
+    :returns: standard output of stat(1) querying the last modification time
+              of ``path``
+    :rtype: str
+    """
+    return stat(session, path, fmt="%z")
+
+
+def get_ctime(session, path):
+    """
+    Query the change time.
+
+    :param session: session to run the command on
+    :type session: ShellSession
+    :param str path: path to the filesystem entry to stat
+    :returns: standard output of stat(1) querying the last change time of
+              ``path``
+    :rtype: str
+    """
+    return stat(session, path, fmt="%y")
+
+
+def get_size(session, path):
+    """
+    Query the file size.
+
+    :param session: session to run the command on
+    :type session: ShellSession
+    :param str path: path to the filesystem entry to stat
+    :returns: standard output of stat(1) querying the size of ``path``
+    :rtype: int
+    """
+    return int(stat(session, path, fmt="%s"))
```

Comment:
On those methods, I can see some advantages on having they separated, so I'm ok with that here.

Reply:
Agreed too, these are advantages that I have kept till the end too.